logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/LogStreamBatchWriterImplTest.java

```diff
@@ -68,6 +68,49 @@ void canWriteAdditionalEvent() {
     verifyNoMoreInteractions(dispatcher);
   }
 
+  /**
+   * This test asserts that {@link LogStreamBatchWriterImpl#canWriteAdditionalEvent(int, int)}
+   * computes the correct batch length before passing it on to the dispatcher to check if it's
+   * acceptable. It also verifies that the same length is passed to the dispatcher when actually
+   * writing, ensuring both methods remain consistent with each other.
+   */
+  @Test
+  void canWriteAdditionalEventWithEventCount() {
```

Comment:
I know you haven't wrote these tests, but I feel they are far to complicated and too coupled with other things. This also causes to fail the tests if we replace dispatcher for example etc. which is an underlying component.

What we actually have changed was the behavior of the `canWriteAdditionalEvent` Can I write a event with value X? Where I shouldn't be able to do so.  Meaning before it returned true, whereas it should return false. This we can add a test for I think.

Reply:
Fully agree on the complexity of the test, we should aim for more focused test going forward on `main` for sure. For the stable branch it might be acceptable though? Let's sync on this!

Reply:
as agreed in the sync, we will keep the tests as is for now, given they are gone on main now